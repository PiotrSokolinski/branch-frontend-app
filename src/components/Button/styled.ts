import styled from "styled-components";

type ContainerProps = {
  width: number;
};

export const Container = styled.div<ContainerProps>`
  padding: 1rem 3rem;
  display: flex;
  align-items: center;
  justify-content: center;
  width: ${({ width, theme }) => theme.rem(width)};
  background-color: ${({ theme }) => theme.colors.java};
  color: ${({ theme }) => theme.colors.blueDianne};
  border-radius: 10rem;
`;

type StyledButtonProps = {
  type?: string;
  disabled?: boolean;
};

export const StyledButton = styled.button.attrs((props: StyledButtonProps) => ({
  type: props.type,
  disabled: props.disabled,
}))<StyledButtonProps>`
  width: fit-content;
  padding: 1rem 3rem;
  outline: none;
  border: none;
  cursor: pointer;
  color: ${({ theme }) => theme.colors.blueDianne};
  ${({ theme }) => theme.fonts.button};
  border-radius: 10rem;
  background-color: ${({ theme, disabled }) =>
    disabled ? theme.colors.aquaHaze : theme.colors.java};

  &:hover {
    background-image: linear-gradient(rgba(0, 0, 0, 0.08), rgba(0, 0, 0, 0.08));
  }
`;
