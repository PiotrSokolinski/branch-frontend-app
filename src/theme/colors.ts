const colors = {
  aquaHaze: "#f4f7f9",
  aquaHazeDark: "#E9EEF3",
  white: "#FFFFFF",
  black: "#000000",
  coralLight: "#FFE7DD",
  coralDark: "#FF7C44",
  java: "#20BFB7",
  javaLight: "#D2F2F1",
  javaDark: "#1CB9B0",
  blueDianne: "#21455B",
  hawkesBlue: "#EDF4FE",
  pigeonPost: "#AECBDD",
  flamingo: "#ef462f",
  tarawera: "#0A354F",
  boulder: "#757575",
  downriver: "#0B2B4A",
  nileBlue: "#1A475F",
  frenchGray: "#BABDC8",
  athensGray: "#E3E6EC",
  zumthor: "#E8F1FF",
  zircon: "#F5F7F9",
  solitude: "#E1EDFF",
  coral: "#FE8856",
  lightGray: "#D3D3D3",
  athensGrayLight: "#F6F7F9",
  bermudaGray: "#6C8B9F",
  tropicalBlue: "#CFDFF9",
  jungleGreen: "#2aaba4",
  linkWater: "#DEE8F5",
  lynch: "#5B7585",
  gray: "#808080",
  iron: "#D0D2D3",
  endeavour: "#005DAA",
  pink: "#DE138B",
  mystic: "#E8EDF2",
  mentation: "#FFA336",
  medication: "#4356FF",
  mobility: "#FA4AE8",
  geyser: "#D3DCDF",
  baliHai: "#90A2AD",
  milanoRed: "#AD1B07",
  EasternBlue: "#1DB0A8",
  viking: "#59DED7",
};

export default colors;
