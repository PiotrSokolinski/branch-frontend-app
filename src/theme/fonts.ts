const fonts = {
  h1: {
    fontFamily: "Poppins",
    fontWeight: "bold",
    fontSize: "5rem",
    lineHeight: "2.7rem",
    letterSpacing: "0.08em",
  },
  h2: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "3rem",
    lineHeight: "1.9rem",
    letterSpacing: "0.08em",
  },
  text: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "1.6rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.08em",
  },
  button: {
    fontFamily: "Poppins",
    fontWeight: "bold",
    fontSize: "1.3rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.08em",
  },
  label: {
    fontFamily: "Poppins",
    fontWeight: "bold",
    fontSize: "1.3rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.02em",
  },
  p: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "1.1rem",
    lineHeight: "1.3rem",
    letterSpacing: "0.02em",
  },
  input: {
    fontFamily: "Poppins",
    fontWeight: "normal",
    fontSize: "1.3rem",
    lineHeight: "1.4rem",
    letterSpacing: "0.08em",
  },
};

export default fonts;
