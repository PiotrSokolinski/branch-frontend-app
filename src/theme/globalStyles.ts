import { createGlobalStyle } from "styled-components";

const GlobalStyles = createGlobalStyle`
  html,
  body {
    font-family: Poppins !important;
    margin: 0;
    padding: 0;
    font-size: 1.3rem;

  }

  * {
    box-sizing: border-box;
  }

  html {
    height: 100%;
    width: 100%;
  }

  body {
    min-height: 100%;
    max-width: 100%;
    display: flex;
    flex: 1;
    align-items: stretch;
    overflow-y: scroll;
    overflow-x: hidden;
  }

  h1 { margin: 0; font-size: 2.1rem; font-weight: bold; }
  h2 { margin: 0; font-size: 1.8rem; font-weight: bold; }
  p { margin: 0; }


  #root {
    display: flex;
    min-height: 100%;
    min-width: 100%;
  }
`;

export default GlobalStyles;
