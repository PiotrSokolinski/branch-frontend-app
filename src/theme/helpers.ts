import { fontSizeMultiplier } from "./config";

export const rem = (px: number): string => `${px / fontSizeMultiplier}rem`;
