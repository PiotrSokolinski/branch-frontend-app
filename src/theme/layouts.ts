const layouts = {
  publicLayout: {
    sidebarWidth: "50vw",
  },
};

export default layouts;
