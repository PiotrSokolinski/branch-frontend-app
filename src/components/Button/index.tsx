import React, { useRef, useState, useEffect } from "react";

import { Container, StyledButton } from "./styled";

type ButtonProps = {
  children: React.ReactChild;
  loading?: boolean;
  disabled?: boolean;
  className?: string;
  onClick?: (e: React.EventHandler<any>) => void;
  type?: string;
};

const DEFAULT_WIDTH = 100;

const Button: React.FC<ButtonProps> = ({
  children,
  loading,
  disabled,
  className,
  onClick,
  type,
}) => {
  const button = useRef(null);
  const [width, setWidth] = useState(DEFAULT_WIDTH);

  useEffect(() => {
    if (button && button.current) {
      // @ts-ignore
      setWidth(button.current.clientWidth);
    }
  }, [loading]);

  if (loading)
    return (
      <Container className={className} width={width}>
        loading...
      </Container>
    );
  return (
    <StyledButton
      ref={button}
      className={className}
      onClick={onClick}
      disabled={disabled}
      type={type}
    >
      {children}
    </StyledButton>
  );
};

export default Button;
