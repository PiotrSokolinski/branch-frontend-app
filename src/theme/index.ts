import { DefaultTheme } from "styled-components";

import colors from "./colors";
import fonts from "./fonts";
import zIndex from "./zIndex";
import layouts from "./layouts";

import { rem } from "./helpers";

const theme: DefaultTheme = {
  colors,
  fonts,
  zIndex,
  layouts,
  rem,
};

export default theme;
