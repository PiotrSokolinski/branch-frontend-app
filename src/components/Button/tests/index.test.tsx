/**
 *
 * Tests for Button
 *
 */

import React from "react";
import { render, fireEvent } from "@testing-library/react";
import { IntlProvider } from "react-intl";

import Button from "../index";
import { ThemeProvider } from "styled-components";
import theme from "../../../theme";

describe("<Button />", () => {
  it("Expect to not log errors in console", () => {
    const spy: jest.SpyInstance<
      void,
      [message?: any, ...optionalParams: any[]]
    > = jest.spyOn(global.console, "error");
    render(
      <ThemeProvider theme={theme}>
        <Button>Button</Button>
      </ThemeProvider>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it("Should render primary button and match the snapshot", () => {
    const { container } = render(
      <ThemeProvider theme={theme}>
        <Button>Button</Button>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
  });

  it("Should invoke onClick callback on the Button", () => {
    const buttonText: string = "Button";
    const onClick: () => void = jest.fn();
    const { getByText } = render(
      <ThemeProvider theme={theme}>
        <Button disabled={false} onClick={onClick}>
          {buttonText}
        </Button>
      </ThemeProvider>
    );
    const button: HTMLElement = getByText(buttonText);

    fireEvent.click(button);

    expect(onClick).toHaveBeenCalledTimes(1);
  });
});
