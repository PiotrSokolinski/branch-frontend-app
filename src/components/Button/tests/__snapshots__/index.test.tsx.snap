// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should render primary button and match the snapshot 1`] = `
<div>
  <button
    class="sc-gsTCUz lkJwrv"
  >
    Button
  </button>
</div>
`;
